llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeWinogradInputPass.cpp

```diff
@@ -0,0 +1,246 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtDialect.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree-dialects/Dialect/LinalgExt/Passes/PassDetail.h"
+#include "iree-dialects/Dialect/LinalgExt/Passes/Passes.h"
+#include "iree-dialects/Dialect/LinalgExt/Utils/Utils.h"
+#include "iree-dialects/Dialect/LinalgExt/Utils/WinogradConstants.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/MemRef/Transforms/Passes.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Dialect/Tensor/Utils/Utils.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/Support/Debug.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace LinalgExt {
+
+namespace {
+
+static void computeLoopParams(SmallVectorImpl<Value> &lbs,
+                              SmallVectorImpl<Value> &ubs,
+                              SmallVectorImpl<Value> &steps, Value tensor,
+                              Location loc, OpBuilder &builder) {
+  auto zero = builder.create<arith::ConstantIndexOp>(loc, 0);
+  auto one = builder.create<arith::ConstantIndexOp>(loc, 1);
+  auto tensorDims = tensor::createDimValues(builder, loc, tensor);
+  for (int i = 2; i < tensorDims.size(); i++) {
+    lbs.push_back(zero);
+    ubs.push_back(getValueOrCreateConstantIndexOp(builder, loc, tensorDims[i]));
+    steps.push_back(one);
+  }
+}
+
+class ReifyWinogradInputTransform final
+    : public OpRewritePattern<WinogradInputTransformOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(WinogradInputTransformOp inputOp,
+                                PatternRewriter &rewriter) const override {
+    auto loc = inputOp.getLoc();
```

Comment:
Nit: I see you pretty much only use `auto` as result types in the function.. We should prefer to specify the result type instead of `auto` if it's not directly obvious.

Reply:
Sounds good. Will change.